tests/foreman/api/test_puppet.py

```diff
@@ -0,0 +1,116 @@
+"""End to end test for Puppet funcionality
+
+@Requirement: Puppet
+
+@CaseAutomation: Automated
+
+@CaseLevel: Acceptance
+
+@CaseComponent: API
+
+@TestType: Functional
+
+@CaseImportance: High
+
+@Upstream: No
+"""
+
+from robottelo.config import settings
+from robottelo.decorators import (
+    run_in_one_thread,
+    run_only_on,
+    skip_if_not_set,
+    stubbed,
+    tier3,
+)
+from robottelo.test import APITestCase
+
+
+@run_in_one_thread
+@skip_if_not_set('clients')
+class PuppetTestCase(APITestCase):
+    """Implements Puppet test scenario"""
+
+    @classmethod
+    def setUpClass(cls):
+        super(PuppetTestCase, cls).setUpClass()
+        cls.sat6_hostname = settings.server.hostname
+
+    @run_only_on('sat')
+    @stubbed
+    @tier3
+    def test_positive_puppet_scenario(self):
+        """
+        @id: 29635f43-e701-4539-844c-275545cdf9c4
```

Comment:
Hey. I think it would be good to add a test description and for all tests below. Sorry I didn't notice earlier.